### ISSUE TITLE
Perform delete on model throws exception

### ISSUE THREAD

**Issue by basherr** [orchestral/tenanti#51]
When you try to perform delete on any `driver` Model, It will throw an exception 
```No query results for model [App\Company] 15```

Since the code for reset migration at [Factory.php](https://github.com/orchestral/tenanti/blob/master/src/Migrator/Factory.php#L120) is responsible for that as 

```
public function reset($database, $id = null, $pretend = false)
    {
        if (! is_null($id)) {
           //while executeById will look for an id which actually does not exist any more
            return $this->executeById($id, function ($entity) use ($database, $pretend) {
                $this->runReset($entity, $database, $pretend);
            });
        }
        $this->executeByChunk(function ($entities) use ($database, $pretend) {
            foreach ($entities as $entity) {
                $this->runReset($entity, $database, $pretend);
            }
        });
    }
```

Code at CompanyController

```
            $company = Company::findOrFail($companyId);
            // $company->manager()->forceDelete();
            $company->delete();
```

**Comment by crynobone** [comment]
Could you try replacing the following method on your observer class.

```php
    /**
     * Run on deleted observer.
     *
     * @param  \Illuminate\Database\Eloquent\Model  $entity
     *
     * @return bool
     */
    public function deleting(Model $entity)
    {
        parent::deleted($entity);
    }

    /**
     * Run on deleted observer.
     *
     * @param  \Illuminate\Database\Eloquent\Model  $entity
     *
     * @return bool
     */
    public function deleted(Model $entity)
    {
        //
    }
```

**Comment by basherr** [comment]
@crynobone Thanks That's just fine

**Comment by crynobone** [comment]
I can this be resolved on `3.5`, but don't think we can introduce a breaking change on `3.4` or below right now.
